File: Candy/js_candy/candy07.js
// 編號：CANDY-007
// 程式語言：JavaScript
// 題目：在某個數字陣列裡，可能藏有某個不合群的奇數或偶數，試著找出它！

function findSomeDifferent(numbers) {
    const evenNum = numbers.filter((n) => n % 2 == 0);
    const oddNum = numbers.filter((n) => n % 2 != 0);
    if (evenNum.length == 1) {
      return evenNum[0];
    } else {
      return oddNum[0];
    };
  };
  
  console.log(findSomeDifferent([2, 4, 0, 100, 4, 11, 2602, 36])) // 印出 11
  console.log(findSomeDifferent([160, 3, 1719, 19, 11, 13, -21])) // 印出 160

File: Candy/rb_candy/candy07.rb
# 編號：CANDY-007
# 程式語言：Ruby
# 題目：在某個數字陣列裡，可能藏有某個不合群的奇數或偶數，試著找出它！

def find_some_different numbers 
  if numbers.count { |num| num.odd? } == 1
    numbers.find { |num| num.odd? }
  else
    numbers.find { |num| num.even? }
  end
end
  
puts find_some_different([2, 4, 0, 100, 4, 11, 2602, 36]) # 印出 11
puts find_some_different([160, 3, 1719, 19, 11, 13, -21]) # 印出 160